pkg/upstream/cluster/loadbalancer.go

```diff
@@ -330,6 +346,77 @@ func newEdfLoadBalancerLoadBalancer(hosts types.HostSet, unWeightChoose func(typ
 	return lb
 }
 
+func slowStartDurationFactorFunc(info types.ClusterInfo, host types.Host) float64 {
+	return slowStartDurationFactorFuncWithNowFunc(info, host, time.Now)
+}
+
+//slowStartDurationFactorFuncWithNowFunc with nowFunc parameter for testing
+func slowStartDurationFactorFuncWithNowFunc(info types.ClusterInfo, host types.Host, nowFunc func() time.Time) float64 {
+	slowStart := info.SlowStart()
+
+	if slowStart.SlowStartWindow <= 0 {
+		return 1.0
+	}
+
+	duration := nowFunc().Sub(host.StartTime())
```

Comment:
If the host's startime always uses the old, the host will only once a slow start.

Reply:
No better solution, I will add comment for that